datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/domain/BatchSetDomainResolverTest.java

```diff
@@ -0,0 +1,344 @@
+    assertTrue(resolver.get(mockEnv).get());
+
+    final Domains newDomains = new Domains().setDomains(new UrnArray(ImmutableList.of(
+        Urn.createFromString(TEST_DOMAIN_2_URN)
+    )));
+
+    final MetadataChangeProposal proposal1 = new MetadataChangeProposal();
+    proposal1.setEntityUrn(Urn.createFromString(TEST_ENTITY_URN_1));
+    proposal1.setEntityType(Constants.DATASET_ENTITY_NAME);
+    proposal1.setAspectName(Constants.DOMAINS_ASPECT_NAME);
+    proposal1.setAspect(GenericRecordUtils.serializeAspect(newDomains));
+    proposal1.setChangeType(ChangeType.UPSERT);
+
+    Mockito.verify(mockService, Mockito.times(1)).ingestProposal(
+        Mockito.eq(proposal1),
+        Mockito.any(AuditStamp.class)
+    );
+
+    final MetadataChangeProposal proposal2 = new MetadataChangeProposal();
+    proposal2.setEntityUrn(Urn.createFromString(TEST_ENTITY_URN_2));
+    proposal2.setEntityType(Constants.DATASET_ENTITY_NAME);
+    proposal2.setAspectName(Constants.DOMAINS_ASPECT_NAME);
+    proposal2.setAspect(GenericRecordUtils.serializeAspect(newDomains));
+    proposal2.setChangeType(ChangeType.UPSERT);
+
+    Mockito.verify(mockService, Mockito.times(1)).ingestProposal(
+        Mockito.eq(proposal2),
+        Mockito.any(AuditStamp.class)
+    );
+
+    Mockito.verify(mockService, Mockito.times(1)).exists(
+        Mockito.eq(Urn.createFromString(TEST_DOMAIN_2_URN))
+    );
+  }
+
+  @Test
+  public void testGetSuccessExistingDomains() throws Exception {
+    final Domains originalDomain = new Domains().setDomains(new UrnArray(ImmutableList.of(
+        Urn.createFromString(TEST_DOMAIN_1_URN))));
+
+    EntityService mockService = Mockito.mock(EntityService.class);
+
+    Mockito.when(mockService.getAspect(
+        Mockito.eq(UrnUtils.getUrn(TEST_ENTITY_URN_1)),
+        Mockito.eq(Constants.DOMAINS_ASPECT_NAME),
+        Mockito.eq(0L)))
+        .thenReturn(originalDomain);
+
+    Mockito.when(mockService.getAspect(
+        Mockito.eq(UrnUtils.getUrn(TEST_ENTITY_URN_2)),
+        Mockito.eq(Constants.DOMAINS_ASPECT_NAME),
+        Mockito.eq(0L)))
+        .thenReturn(originalDomain);
+
+    Mockito.when(mockService.exists(Urn.createFromString(TEST_ENTITY_URN_1))).thenReturn(true);
+    Mockito.when(mockService.exists(Urn.createFromString(TEST_ENTITY_URN_2))).thenReturn(true);
+
+    Mockito.when(mockService.exists(Urn.createFromString(TEST_DOMAIN_1_URN))).thenReturn(true);
+    Mockito.when(mockService.exists(Urn.createFromString(TEST_DOMAIN_2_URN))).thenReturn(true);
+
+    BatchSetDomainResolver resolver = new BatchSetDomainResolver(mockService);
+
+    // Execute resolver
+    QueryContext mockContext = getMockAllowContext();
+    DataFetchingEnvironment mockEnv = Mockito.mock(DataFetchingEnvironment.class);
+    BatchSetDomainInput input = new BatchSetDomainInput(TEST_DOMAIN_2_URN, ImmutableList.of(
+        new ResourceRefInput(TEST_ENTITY_URN_1, null, null),
+        new ResourceRefInput(TEST_ENTITY_URN_2, null, null)));
```

Comment:
relevant test to previous question on if we should allow them to add a domain to an asset that already has a domain

Reply:
Yeah so you can change (or unset a domain)